prompts/ddi.py

```diff
@@ -0,0 +1,193 @@
+"""
+DDI Corpus
+
+María Herrero-Zazo, Isabel Segura-Bedmar, Paloma Martínez, Thierry Declerck, The DDI corpus:
+An annotated corpus with pharmacological substances and drug–drug interactions, Journal of Biomedical Informatics,
+Volume 46, Issue 5, October 2013, Pages 914-920, http://dx.doi.org/10.1016/j.jbi.2013.07.011.
+
+Usage:
+python ddi.py --outdir biomedical/
+
+"""
+import os
+import argparse
+from pathlib import Path
+from loguru import logger
+from functools import partial
+from typing import List, Callable
+from pybrat.parser import BratParser, Example
+from utils import download, uncompress
+from prompts import DatasetPrompts
+
+# TODO: Implement proper Hugging Face's Datasets version of DDICorpus
+_HOMEPAGE = "https://github.com/isegura/DDICorpus"
+_URL = "https://github.com/isegura/DDICorpus/raw/master/DDICorpus-2013(BRAT).zip"
+_TRAINING_PATH = "Train"
+_TEST_PATH = "Test"
+_BRAT_PARSER = BratParser(error="ignore")
+
+
+def load_ddi_corpus_split(path: str) -> List[Example]:
+    logger.info(f"Loading DDICorpus split from {path}")
+    return _BRAT_PARSER.parse(path)
+
+
+class DDICorpusPrompts(DatasetPrompts):
+    """Prompts creator from DDI Corpus."""
+
+    def __init__(self, data_root: str):
+        """Instantiate DDICorpusPrompts.
+
+        Args:
+            data_root: dataset download root folder.
+        """
+        self.data_root = Path(data_root).resolve()
+        self.path = self.data_root / _URL.split("/")[-1]
+        self._init_dataset()
+        self._init_prompts()
+
+    def _init_dataset(self):
+        """Download dataset and create splits."""
+        # confirm dataset dir exist
+        if not os.path.exists(self.data_root):
+            os.makedirs(self.data_root)
+            logger.info(f"Directory {self.data_root} successfully created")
+
+        # download dataset file
+        if not os.path.exists(self.path):
+            download(_URL, self.path)
+            logger.info("Dataset successfully downloaded")
+
+        # uncompress file
+        full_path = self.data_root / "DDICorpusBrat"
+        if not os.path.exists(full_path):
+            uncompress(self.path, self.data_root)
+
+        self.splits = {
+            split: load_ddi_corpus_split(str(full_path / path))
+            for split, path in {"train": _TRAINING_PATH, "test": _TEST_PATH}.items()
+        }
+
+    def _init_prompts(self):
+        """Initialize prompts."""
+        self._prompts = {}
+        self._metadata = {}
+
+
+# NOTE: implemented prompts
+def list_entity_template(
```

Comment:
This is great!! Much more useful and generic that my implementation! Ideally we should start building a set of generic prompting functions like this (and for the relational/QA transformations) in a separate module to use across datasets/tasks.

Reply:
Hey @jason-fries glad you found this useful. If you want I can also spend sometime packaging the whole folder as a python module and refactoring a bit.